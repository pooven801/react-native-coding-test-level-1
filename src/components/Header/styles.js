import { StyleSheet } from "react-native";

export default StyleSheet.create({
  buttonContainer: {
    backgroundColor: "#7a8b8b",
    width: 80,
    height: 30,
    margin: 20,
    borderRadius: 5,
    alignItems: "center",
    justifyContent: "center",
  },

  textStyles: {
    color: "white",
    fontWeight: "bold",
    fontSize: 16,
  },
});
