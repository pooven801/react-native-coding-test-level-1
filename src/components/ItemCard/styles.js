import { StyleSheet, Dimensions } from "react-native";

export default StyleSheet.create({
  container: {
    width: "90%",
    height: Dimensions.get("window").width / 2,
    backgroundColor: "#525c5d",
    alignSelf: "center",
    borderRadius: 20,
    marginBottom: 10,
  },

  titleStyle: {
    fontSize: 30,
    alignSelf: "center",
    color: "white",
    paddingTop: 20,
  },

  buttonStyle: {
    width: "90%",
    height: 45,
    backgroundColor: "#8fa2a2",
    borderRadius: 10,
    alignSelf: "center",
    position: "absolute",
    bottom: 20,
    alignItems: "center",
    justifyContent: "center",
  },

  buttonTextStyle: { fontSize: 24, color: "white" },
});
