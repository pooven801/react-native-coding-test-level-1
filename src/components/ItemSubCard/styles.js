import { StyleSheet } from "react-native";

export default StyleSheet.create({
  itemContainer: {
    width: "90%",
    backgroundColor: "#7a8b8b",
    marginBottom: 10,
    minHeight: 30,
    alignSelf: "center",
    borderRadius: 10,
  },

  titleStyle: {
    fontSize: 20,
    fontWeight: "bold",
    padding: 10,
    color: "#e6f3f4",
  },

  subTitleStyle: {
    fontSize: 18,
    paddingLeft: 15,
    paddingBottom: 5,
    color: "#e6f3f4",
  },
});
