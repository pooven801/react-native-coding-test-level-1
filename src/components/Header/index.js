import React from "react";
import {
    View,
    TouchableOpacity,
    Text
} from "react-native";
import styles from "./styles";
import PropTypes from "prop-types";

export default function Header(props) {
    const {
        onPress
    } = props;


    return (
        <TouchableOpacity
            onPress={onPress}
            style={styles.buttonContainer}>
            <Text style={styles.textStyles}>Back</Text>
        </TouchableOpacity>
    )
}

Header.propTypes = {
    onPress: PropTypes.func,

};

Header.defaultProps = {
    onPress: () => { },
};
