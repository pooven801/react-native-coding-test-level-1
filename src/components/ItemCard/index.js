import React from "react";
import { View, TouchableOpacity, Text } from "react-native";
import styles from "./styles";
import PropTypes from "prop-types";

export default function ItemCard(props) {
  const { onPress, pokemonName, showButton } = props;

  return (
    <View style={styles.container}>
      <Text style={styles.titleStyle}>{pokemonName}</Text>
      {showButton && (
        <TouchableOpacity onPress={onPress} style={styles.buttonStyle}>
          <Text style={styles.buttonTextStyle}>Show Detail</Text>
        </TouchableOpacity>
      )}
    </View>
  );
}

ItemCard.propTypes = {
  onPress: PropTypes.func,
  pokemonName: PropTypes.string,
  showButton: PropTypes.bool,
};

ItemCard.defaultProps = {
  onPress: () => {},
  pokemonName: "",
  showButton: false,
};
