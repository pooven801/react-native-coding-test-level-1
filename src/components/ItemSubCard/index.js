import React from "react";
import { View, Text } from "react-native";
import styles from "./styles";
import PropTypes from "prop-types";

export default function ItemSubCard(props) {
  const { title, subTitle } = props;
  return (
    <View style={styles.itemContainer}>
      <Text style={styles.titleStyle}>{title}</Text>
      {subTitle != [] &&
        subTitle.map((item, index) => {
          return (
            <Text
              key={index}
              style={[
                styles.subTitleStyle,
                index + 1 == subTitle.length && { paddingBottom: 10 },
              ]}
            >
              {"\u2B24 " + item}
            </Text>
          );
        })}
    </View>
  );
}

ItemSubCard.propTypes = {
  title: PropTypes.string,
  subTitle: PropTypes.array,
};

ItemSubCard.defaultProps = {
  title: "",
  subTitle: [],
};
