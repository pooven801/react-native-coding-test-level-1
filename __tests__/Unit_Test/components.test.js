import {create} from 'react-test-renderer';
import Header from '../../src/components/Header';
import ItemCard from '../../src/components/ItemCard';
import ItemSubCard from '../../src/components/ItemSubCard';


const treeHeader = create(<Header />);
const treeItemCard = create(<ItemCard />);
const treeItemSubCard = create(<ItemSubCard />);


test('Snapshot: Component > Header', () => {
    expect(treeHeader).toMatchSnapshot()
})

test('Snapshot: Component > ItemCard', () => {
    expect(treeItemCard).toMatchSnapshot()
})

test('Snapshot: Component > ItemSubCard', () => {
    expect(treeItemSubCard).toMatchSnapshot()
})