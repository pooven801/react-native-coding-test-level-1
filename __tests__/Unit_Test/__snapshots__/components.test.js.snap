// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot: Component > Header 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#7a8b8b",
      "borderRadius": 5,
      "height": 30,
      "justifyContent": "center",
      "margin": 20,
      "opacity": 1,
      "width": 80,
    }
  }
>
  <Text
    style={
      {
        "color": "white",
        "fontSize": 16,
        "fontWeight": "bold",
      }
    }
  >
    Back
  </Text>
</View>
`;

exports[`Snapshot: Component > ItemCard 1`] = `
<View
  style={
    {
      "alignSelf": "center",
      "backgroundColor": "#525c5d",
      "borderRadius": 20,
      "height": 375,
      "marginBottom": 10,
      "width": "90%",
    }
  }
>
  <Text
    style={
      {
        "alignSelf": "center",
        "color": "white",
        "fontSize": 30,
        "paddingTop": 20,
      }
    }
  >
    
  </Text>
</View>
`;

exports[`Snapshot: Component > ItemSubCard 1`] = `
<View
  style={
    {
      "alignSelf": "center",
      "backgroundColor": "#7a8b8b",
      "borderRadius": 10,
      "marginBottom": 10,
      "minHeight": 30,
      "width": "90%",
    }
  }
>
  <Text
    style={
      {
        "color": "#e6f3f4",
        "fontSize": 20,
        "fontWeight": "bold",
        "padding": 10,
      }
    }
  >
    
  </Text>
</View>
`;
